contracts/optimistic-ethereum/OVM/execution/OVM_ExecutionManager.sol

```diff
@@ -1835,11 +1836,24 @@ contract OVM_ExecutionManager is iOVM_ExecutionManager, Lib_AddressResolver {
 
         ovmStateManager = _ovmStateManager;
         _initContext(_transaction);
-
         messageRecord.nuisanceGasLeft = uint(-1);
-        messageContext.ovmADDRESS = _transaction.entrypoint;
-        messageContext.ovmCALLER = _from;
 
-        return _transaction.entrypoint.call{gas: _transaction.gasLimit}(_transaction.data);
+        messageContext.ovmADDRESS = _from;
+
+        bool isCreate = _transaction.entrypoint == address(0);
+        if (isCreate) {
+            (address created, bytes memory revertData) = ovmCREATE(_transaction.data);
+            if (created == address(0)) {
+                return (false, revertData);
+            } else {
+                return (true, Lib_EthUtils.getCode(created));
```

Comment:
Hmmm is `getCode` necessary here?

Reply:
It was the easiest way I saw to do it.  The return type is correct if that's what you're getting at -- the `eth_call` endpoint for `to = undefined` will return the deployed bytecode if successful, differing from a standard evmCREATE message.

Reply:
Oh interesting. I wasn't aware of that. Sounds good.

Reply:
Actually: could we put a comment here explaining just that? Even copy/paste what you wrote here would be fine.